### PR TITLE
Implement static slug for `Contact` page and add more tests to verify the changes

### LINKED ISSUE #289: Static slug for contact detail page
## Description

Related with #288 but for Contact Detail Page.

Currently, we generate a slug based on the index and this can be changed everytime.

We need to set a static slug for this.

See the detail on the Doc on section 1.2.3:

https://docs.google.com/document/d/1im7CF8XaF4yYQ_jJjjW7dOKZz8tJFvidtW7bOlqt1Y8/edit?usp=sharing

## Tasks

- [ ] Generate slug from the `etc/fetcher`, slugify the title, e.g: `3R Medika Gas` will be `3r-medika-gas`
- [ ] Use the slug on the client side, without any additional index
- [ ] Test the functionality

### LINKED ISSUE #383: Write some unit & integration tests for `Contact` page to avoid regressions

### PR DESCRIPTION
**Pull request by zainfathoni**
Closes #383
Closes  #289 because it's hard to write the tests without closing it

## Current Tasks

- [x] renders the title and the breadcrumbs in ContactPage correctly
- [x] test getStaticPaths
- [x] remove the additional last segment from the contact slug
- [x] test getStaticProps

### CODE REVIEW

**Comment by mazipan** [comment]
https://deploy-preview-469--wargabantuwarga.netlify.app/provinces/dki-jakarta?kebutuhan=Bantuan%20kebutuhan%20pokok

We have an edge case, when the data from 1 penyedia but submit multiple times since we still not support multiple "keterangan".

The problem, when we open any links, it will return the same "keterangan"

Screenshot_20210727-175632_Chrome

**Comment by zainfathoni** [comment]
I see, so it should be a permutation of penyedia & keterangan.

```
`${penyedia}-${keterangan}`
```

Do you think it's okay or not, Mas @mazipan?

**Comment by mazipan** [comment]
Screenshot_20210727-180914_Chrome

This also, https://deploy-preview-469--wargabantuwarga.netlify.app/provinces/dki-jakarta?kebutuhan=Kontak%20penting

Some data have same "Penyedia" and "Keterangan" 😂
The only unique part is the telephone

**Comment by mazipan** [comment]
The ideal way, is we set the unique ID on the Google Sheets for each row.
So we combine it with the ID from the Google Sheets, not from the index we generate.

**Comment by mazipan** [comment]
> I see, so the slug should be a permutation of penyedia & keterangan.
> 
> ```
> slug = `${penyedia}-${keterangan}`
> ```
> 
> Do you think it's okay or not, Mas @mazipan ?
> 
> If it's okay, I'll implement it and write the tests for the edge cases.

For the edge case no.1, yes.
But for no.2 NO

**Comment by mazipan** [comment]
The `StaticPaths` for the Next.js in the Contact Detail Page should generate by the fetcher.

We don't need to reinvent, looping the data, get the slug, to get the staticPaths available.

**Review by resir014 — APPROVED** [review]

**Comment by zainfathoni** [comment]
> The ideal way, is we set the unique ID on the Google Sheets for each row.
> 
> So we combine it with the ID from the Google Sheets, not from the index we generate by our own script.
> 
> 
> 
> Imagine it as an ID on the database
> 
> 
> 
> I think we need to connect with the team that handled the Google Sheets content to discuss in how we set an unique ID for each row.
> 

I think we cannot rely on manual efforts like that to get the ID.

> 
> Or am I missing? is there any technical workaround to get an unique ID per row per tab on the Google Sheets?

We can append the kebutuhan & phone number into the mix, because if they have the same Penyedia, Keterangan, Phone Number, & Kebutuhan, it means that they should have been merged instead. 😅

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "lint": "eslint \"**/*.{js,jsx,ts,tsx}\"",
     "lint:fix": "eslint --fix \"**/*.{js,jsx,ts,tsx}\"",
     "mirror-box": "ts-node etc/mirror-box.ts",
-    "netlify-export": "yarn mirror-box && yarn build && next export",
+    "netlify-export": "yarn fetch-wbw && yarn build && next export",
```
Comment (zainfathoni) [inline package.json:15]: @adityapurwa, please pull the latest `main` branch to your [wbw-box](https://wbw-box.lucentshard.com/) instance once this PR is merged, thanks! 🙏
